cpu/stm32/kconfigs/Kconfig.clk

```diff
@@ -174,31 +153,104 @@ config CLOCK_PLL_MUL
     default 24 if PLL_MUL_24
     default 32 if PLL_MUL_32
     default 48 if PLL_MUL_48
+endif  # CPU_FAM_L0 || CPU_FAM_L1
 
+if CPU_FAM_G0
+choice
+bool "HSISYS division factor" if USE_CLOCK_HSI
+default CLOCK_HSISYS_DIV_1
+
+config CLOCK_HSISYS_DIV_1
+    bool "Divide HSISYS by 1"
+
+config CLOCK_HSISYS_DIV_2
+    bool "Divide HSISYS by 2"
+
+config CLOCK_HSISYS_DIV_4
+    bool "Divide HSISYS by 4"
+
+config CLOCK_HSISYS_DIV_8
+    bool "Divide HSISYS by 8"
+
+config CLOCK_HSISYS_DIV_16
+    bool "Divide HSISYS by 16"
+
+config CLOCK_HSISYS_DIV_32
+    bool "Divide HSISYS by 32"
+
+config CLOCK_HSISYS_DIV_64
+    bool "Divide HSISYS by 64"
+
+config CLOCK_HSISYS_DIV_128
+    bool "Divide HSISYS by 128"
+
+endchoice
+
+config CLOCK_HSISYS_DIV
+    int
+    default 1 if CLOCK_HSISYS_DIV_1
+    default 2 if CLOCK_HSISYS_DIV_2
+    default 4 if CLOCK_HSISYS_DIV_4
+    default 8 if CLOCK_HSISYS_DIV_8
+    default 16 if CLOCK_HSISYS_DIV_16
+    default 32 if CLOCK_HSISYS_DIV_32
+    default 64 if CLOCK_HSISYS_DIV_64
+    default 128 if CLOCK_HSISYS_DIV_128
+endif  # CPU_FAM_G0
+
+if CPU_FAM_L0 || CPU_FAM_L1 || CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 choice
-bool "Desired MSI clock frequency" if USE_CLOCK_MSI
-default CLOCK_MSI_4MHZ
+bool "Desired MSI clock frequency" if USE_CLOCK_MSI || (USE_CLOCK_PLL && CLOCK_PLL_SRC_MSI)
+default CLOCK_MSI_48MHZ if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
+default CLOCK_MSI_4MHZ if CPU_FAM_L0 || CPU_FAM_L1
 
 config CLOCK_MSI_65KHZ
-    bool "65.536kHz"
+    bool "65.536kHz" if CPU_FAM_L0 || CPU_FAM_L1
+
+config CLOCK_MSI_100KHZ
+    bool "100kHz" if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 
 config CLOCK_MSI_130KHZ
-    bool "131.072kHz"
+    bool "131.072kHz" if CPU_FAM_L0 || CPU_FAM_L1
+
+config CLOCK_MSI_200KHZ
+    bool "200kHz" if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 
 config CLOCK_MSI_260KHZ
-    bool "262.144kHz"
+    bool "262.144kHz" if CPU_FAM_L0 || CPU_FAM_L1
+
+config CLOCK_MSI_400KHZ
+    bool "400kHz" if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 
 config CLOCK_MSI_520KHZ
-    bool "524.288kHz"
+    bool "524.288kHz" if CPU_FAM_L0 || CPU_FAM_L1
+
+config CLOCK_MSI_800KHZ
+    bool "800kHz" if CPU_FAM_L4 || CPU_FAM_L5 || CPU_FAM_WB
 
 config CLOCK_MSI_1MHZ
-    bool "1.048MHz"
+    bool "1MHz"
```

Comment:
```suggestion
    bool
    prompt "1.048MHz" if CPU_FAM_L0 || CPU_FAM_L1
    prompt "1MHz"
```

Reply:
Hmm sorry, it seems we will need to change these prompts back. Apparently only one prompt is accepted in a simple place, even when they have conditionals, and this is triggering a warning. From kconfiglib:
>        # 'prompt' properties override each other within a single definition of
>        # a symbol, but additional prompts can be added by defining the symbol
>        # multiple times